### ISSUE TITLE
Linux 6.10 compatibility on s390x

### ISSUE THREAD

**Issue by Harry-Chen** [openzfs/zfs#16361]
This issue tracks the compatibility with Linux 6.10 on IBM s390x arch. The main problem originates from the GPL only symbol `vm_layout` added in Linux 6.10-rc [1](https://lore.kernel.org/lkml/Zjn312L7JK2rA8v%2F@tuxmaker.boeblingen.de.ibm.com/) [2](https://github.com/torvalds/linux/commit/236f324b747370b97030c9582591f459353e3589) (and some following commits), which would cause the following error:

```text
  CC [M]  /home/ubuntu/zfs/module/os/linux/zfs/zvol_os.o
  LD [M]  /home/ubuntu/zfs/module/zfs.o
  MODPOST /home/ubuntu/zfs/module/Module.symvers
ERROR: modpost: GPL-incompatible module zfs.ko uses GPL-only symbol 'vm_layout'
make[5]: *** [scripts/Makefile.modpost:145: /home/ubuntu/zfs/module/Module.symvers] Error 1
make[4]: *** [/usr/src/linux-headers-6.10.0-13-generic/Makefile:1891: modpost] Error 2
```

The original comment is made by @piso77 in https://github.com/openzfs/zfs/pull/16250#issuecomment-2202351855

This problem is also reported downstream in Ubuntu: https://bugs.launchpad.net/ubuntu/+source/zfs-linux/+bug/2071774

TL;DR: kernel s390 arch maintainer Vasily Gorbik has proposed a patch to change `vm_layout` to non-GPL-only. It will land in Linux 6.11 and (hopefully) 6.10 stable.

**Comment by tonyhutter** [comment]
@Harry-Chen thank you for the info!  Looks like s390 ZFS users may want to skip the 6.10 kernels then.

**Comment by tjaalton** [comment]
this is in 6.10.3

**Comment by Harry-Chen** [comment]
@piso77 Would you have another try on 6.10.3?

**Comment by tjaalton** [comment]
builds fine with 6.11rc at least
https://autopkgtest.ubuntu.com/packages/z/zfs-linux/oracular/s390x
